app/jobs/calculate_distributions_job.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class CalculateDistributionsJob < ApplicationJob
+  queue_as :default
+
+  def perform
+    clean_up_redis
+    Questionnaire.find_each do |questionnaire|
+      CalculateDistribution.run!(questionnaire: questionnaire)
+    end
+  end
+
+  def clean_up_redis
+    RedisService.keys.each do |key|
+      next unless key.match?(/^distribution_/)
+
+      RedisService.del(key) unless Questionnaire.where(key: key[13..]).count.positive?
```

Comment:
why 13? Prehaps add comment or constant

Reply:
It's because it's the length of `distribution_`

Reply:
I added a comment.